core/decorators/focus-tests-decorator.ts

```diff
@@ -1,8 +1,8 @@
 import "reflect-metadata";
 import { FOCUS } from "./_metadata-keys";
 
-export function FocusTests(constructor: Function) {
+export function FocusTests(constructor: ({ new (...args: Array<any>): any; }) | (() => any)) {
```

Comment:
@Jameskmonger shouldn't this always be a constructor? What's the reason for this type here?